src/state/application/hooks.ts

```diff
@@ -88,9 +88,22 @@ export function useToggleModal(modal: ApplicationModal): () => void {
   return useCallback(() => dispatch(setOpenModal(isOpen ? null : modal)), [dispatch, modal, isOpen])
 }
 
-export function useCloseModal(): () => void {
+export function useCloseModal() {
   const dispatch = useAppDispatch()
-  return useCallback(() => dispatch(setOpenModal(null)), [dispatch])
+  const currentlyOpenModal = useAppSelector((state: AppState) => state.application.openModal)
+  return useCallback(
+    (modal?: ApplicationModal) => {
+      // Check if modal is defined and either matches the currently open modal
+      // or is not a valid instance of ApplicationModal enum (i.e., if it is passed a click event)
+      if ((modal && currentlyOpenModal === modal) || (modal && !Object.values(ApplicationModal).includes(modal))) {
+        dispatch(setOpenModal(null))
+      } else if (!modal) {
```

Comment:
nit: make this the first check so you don't have to check in both places in the other branch